Rebase 1.3.3

uajain opened the pull request:
Summary of changes:
pick <commit_id> (commit message) (rebase-note)

pick 7b58a0bb Checkout using the same mode of the repo (clean)
pick 6b92ce17 build-finish: also export speedwagon files (clean)
pick ff5eb142 build-finish: Add Endless Discovery Feed Providers (clean)
pick a136def8 Allow defining an external location for the system installation and use it (minor conflicts)
pick 3a736c7f Include the external installation directory in XDG_DATA_DIRS (clean)
pick a1922efa flatpak-dir: skip rewriting search provider .ini files for Endless apps (clean)
pick 535f080e common/dir: Change default of core.add-remotes-config-dir to false (clean)
pick 0bdc8108 profile/flatpak.sh: add export/bin dirs to PATH (minor conflicts)

pick 90f1af2e Support migration from old app-ids to new app-id (drop in favour of [upstream commit](https://github.com/flatpak/flatpak/commit/e17b2635d686))
pick 67706810 run: Expose previous-ids symlinks into sandbox (drop in favour of [upstream commit](https://github.com/flatpak/flatpak/commit/e17b2635d686))
(If I am not wrong this is taken care by `flatpak_context_append_bwrap_filesystem` by appending
previous_app_id_dirs to this function)


pick d9cb6fa3 dir: add previous-ids to exported .desktop files  (dropped as [upstream](https://github.com/flatpak/flatpak/commit/a9462d80c86ce2))
pick ba0f678a Test preserving previous-ids and X-Flatpak-RenamedFrom (drop as [upstreamed](https://github.com/flatpak/flatpak/commit/113ee117c058f09db872d6aa83169c3a6c639bd0))
pick 06da7fbe flatpak-run: Add parental controls support for filtering apps (minor conflict)
pick fec038a2 flatpak-run: Use synchronous parental controls method (squashed in 06da7fbe)
pick c44981c1 flatpak-run: Ignore missing app filters for system users (clean)
pick e142e6b9 Revert "Move flatpak_dir_load_appstream_store" (minor conflicts)
pick 259c3333 dir: Support filtering app installs by user’s OARS settings (clean)
pick 344e95c4 dir: Allow app filter to disallow app installation entirely (clean)
pick 7fff9914 system-helper: Move parental controls checks to system-helper (minor conflicts)
pick a09bd091 system-helper: Don’t try and get parental controls for non-apps/runtimes (clean)
pick 58df095f dir: improve error when an app's content is too extreme (clean)
pick 883c8f82 doc: Mention parental controls in flatpak-run documentation (clean)
pick d4dbee59 flatpak-run: mark parental controls message for translation (squashed in 06da7fbe)
pick 11a5bf5b Revert "Stop linking libflatpak against listappstream-glib" (clean)
pick a1266f6c Fix a compiler warning in use of flatpak_dir_new_deploy_data() (drop)
pick da1100f2 system-helper: Properly use dir_get_system() (squashed into 7fff9914)
pick c0174873 dir: Avoid parental controls check in testlibrary (clean)
pick 8999b8ea Consistently use flatpak_load_http_uri() (dropped as cherrypicked commit from upstream)
pick 8341a39d transaction: Don't check dependency installations' remotes (dropped as commit cherrypicked from upstream)





https://phabricator.endlessm.com/T26434

mwleeds commented:
The commit "system-helper: Properly use dir_get_system()" can be squashed into the commit "system-helper: Move parental controls checks to system-helper". No reason to keep it separate

mwleeds commented:
> pick 0bdc810 profile/flatpak.sh: add export/bin dirs to PATH(clean) (dropped as already upstreamed)

I don't think that commit is upstream. The downstream commit modifies `$PATH`, whereas the upstream commit you linked to changes whether or not we modify `$XDG_DATA_DIRS`.

mwleeds commented:
> pick 6770681 run: Expose previous-ids symlinks into sandbox (drop in favour of upstream commit)
(If I am not wrong this is taken care by flatpak_context_append_bwrap_filesystem by appending
previous_app_id_dirs to this function)

Yes I think you're correct

uajain commented:
> The commit "system-helper: Properly use dir_get_system()" can be squashed into the commit "system-helper: Move parental controls checks to system-helper". No reason to keep it separate.

Addressed.

> > pick [0bdc810](https://github.com/endlessm/flatpak/commit/0bdc8108ebd56062a7c40b767a968eb03350e24d) profile/flatpak.sh: add export/bin dirs to PATH(clean) (dropped as already upstreamed)
> 
> I don't think that commit is upstream. The downstream commit modifies `$PATH`, whereas the upstream commit you linked to changes whether or not we modify `$XDG_DATA_DIRS`.

Ah yes, you are correct. I misread the downstream commit. Addressed in the latest iteration of the branch.

mwleeds commented:
It's a nitpick but for the commit "flatpak-run: Add parental controls support for filtering apps", can you indent the second line in the note about the rebase, so it's a bit easier to read?

mwleeds commented:
In my opinion "flatpak-run: Use synchronous parental controls method" should be squashed into the previous commit.

mwleeds commented:
For the commit "Revert "Move flatpak_dir_load_appstream_store"" I don't think it's accurate to say you're adding the commit reversion since you're just rebasing it.

mwleeds commented:
> It's a nitpick but for the commit "flatpak-run: Add parental controls support for filtering apps", can you indent the second line in the note about the rebase, so it's a bit easier to read?

Similarly for "system-helper: Move parental controls checks to system-helper"

mwleeds commented on system-helper/flatpak-system-helper.c:
```diff
@@ -1965,15 +2057,37 @@ flatpak_authorize_method_handler (GDBusInterfaceSkeleton *interface,
+          return FALSE;
+        }
+
+      if (!authorize_deploy_add_polkit_details (installation, invocation, ref, remote,
```
It's strange that we're adding the details here about whether the app is appropriate for the user but in `org.freedesktop.Flatpak.rules.in` the details are only checked if the action is `app-install` not `install-bundle`. So unless I'm missing something, the user could install a bundle that is "too extreme" for them.

mwleeds replied:
If that is a bug it predates the rebase, but perhaps still worth looking into.

uajain replied:
I recommend filing a subtask ticket for this and track it there. I am not very sure I will have time to look into this week.

wjt replied:
@pwithnall is in the process of submitting this code upstream https://github.com/flatpak/flatpak/pull/2797 so please coordinate with him.

mwleeds replied:
I moved the comment to that PR.

mwleeds commented:
It would be easy to squash "flatpak-run: mark parental controls message for translation" into "flatpak-run: Add parental controls support for filtering apps"

mwleeds commented on common/Makefile.am.inc:
```diff
@@ -142,9 +142,10 @@ libflatpak_common_la_CFLAGS = \
 	-I$(srcdir)/dbus-proxy \
 	$(NULL)
-libflatpak_common_la_LIBADD = $(AM_LIBADD) libglnx.la $(BASE_LIBS) $(OSTREE_LIBS) $(SOUP_LIBS) $(JSON_LIBS) $(XAUTH_LIBS) $(LIBSECCOMP_LIBS) $(INTERNAL_GPGME_LIBS) $(SYSTEMD_LIBS) $(DCONF_LIBS)
+libflatpak_common_la_LIBADD = $(AM_LIBADD) libglnx.la $(APPSTREAM_GLIB_LIBS) $(BASE_LIBS) $(OSTREE_LIBS) $(SOUP_LIBS) $(JSON_LIBS) $(XAUTH_LIBS) $(LIBSECCOMP_LIBS) $(INTERNAL_GPGME_LIBS) $(SYSTEMD_LIBS) $(DCONF_LIBS) $(EOS_PARENTAL_CONTROLS_LIBS)
```
This is my fault because it was introduced in the last rebase, but this commit should only be adding `EOS_PARENTAL_CONTROLS_LIBS` not `APPSTREAM_GLIB_LIBS` and the latter should be added by the commit "Revert "Stop linking libflatpak against listappstream-glib""

uajain replied:
Rectified in the latest iteration and mentioned the rectification in relevant commit's message.

mwleeds commented:
I made one change which I don't think requires a review cycle: in `profile/flatpak.sh` I moved the block that modifies `$PATH` to inside the if block, so it only gets executed if flatpak is installed. Of course an Endless system will never be without flatpak, but it seems more correct.
